src/test/java/org/lsc/plugins/connectors/james/JamesIdentityDstServiceTest.java

```diff
@@ -0,0 +1,456 @@
+package org.lsc.plugins.connectors.james;
+
+import static io.restassured.RestAssured.requestSpecification;
+import static io.restassured.RestAssured.with;
+import static io.restassured.config.EncoderConfig.encoderConfig;
+import static io.restassured.config.RestAssuredConfig.newConfig;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.SoftAssertions.assertSoftly;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.core.Is.is;
+import static org.lsc.plugins.connectors.james.TMailContactDstService.FIRSTNAME_KEY;
+import static org.lsc.plugins.connectors.james.TMailContactDstService.SURNAME_KEY;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.FileReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.security.Security;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.time.Instant;
+import java.util.Date;
+import java.util.Map;
+
+import org.apache.commons.lang.RandomStringUtils;
+import org.apache.http.HttpStatus;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.openssl.PEMReader;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.lsc.LscDatasetModification;
+import org.lsc.LscDatasets;
+import org.lsc.LscModificationType;
+import org.lsc.LscModifications;
+import org.lsc.beans.IBean;
+import org.lsc.configuration.PluginConnectionType;
+import org.lsc.configuration.PluginDestinationServiceType;
+import org.lsc.configuration.ServiceType.Connection;
+import org.lsc.configuration.TaskType;
+import org.lsc.plugins.connectors.james.generated.JamesIdentityService;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.utility.MountableFile;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import io.restassured.RestAssured;
+import io.restassured.builder.RequestSpecBuilder;
+import io.restassured.http.ContentType;
+
+class JamesIdentityDstServiceTest {
+    public static final String DOMAIN = "james.org";
+    public static final String BOB = "bob@" + DOMAIN;
+    public static final String ALICE = "alice@" + DOMAIN;
+    private static final URL PRIVATE_KEY = ClassLoader.getSystemResource("conf/jwt_privatekey");
+    private static final URL PUBLIC_KEY = ClassLoader.getSystemResource("conf/jwt_publickey");
+    private static final int JAMES_WEBADMIN_PORT = 8000;
+    private static final boolean FROM_SAME_SERVICE = true;
+
+    private GenericContainer<?> james;
+    private JamesIdentityDstService testee;
+
+    @BeforeEach
+    void setup() throws Exception {
+        james = new GenericContainer<>("quanth99/tmail-backend-memory:lsc-identity-2");
```

Comment:
Let's not forget to put it back to master when the necessary stuff has been merged :)

Reply:
Pro tip: => Create a dedicated issue in the sprint backlog